Sources/MongoSwift/MongoError.swift

```diff
@@ -137,18 +163,39 @@ public struct BulkWriteError: Codable {
 
     private enum CodingKeys: String, CodingKey {
         case code
+        case codeName
         case message = "errmsg"
         case index
     }
+
+    // TODO: can remove this once SERVER-36755 is resolved
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.code = try container.decode(ServerErrorCode.self, forKey: .code)
+        self.message = try container.decode(String.self, forKey: .message)
+        self.index = try container.decode(Int.self, forKey: .index)
+        self.codeName = try container.decodeIfPresent(String.self, forKey: .codeName) ?? ""
+    }
+
+    // TODO: can remove this once SERVER-36755 is resolved
+    public init(code: ServerErrorCode, codeName: String, message: String, index: Int) {
```

Comment:
I think this one can also become `internal`

Reply:
whoops, also fixed